packages/axelar-wasm-std/src/voting.rs

```diff
@@ -154,13 +153,12 @@ pub enum PollStatus {
 
 #[cw_serde]
 pub struct WeightedPoll {
-    poll_id: PollID,
-    snapshot: Snapshot,
-    expires_at: u64,
-    poll_size: u64,
-    votes: Vec<Uint256>, // running tally of weighted votes
-    status: PollStatus,
-    voted: HashSet<Addr>,
```

Comment:
I'm not sure how I feel about removing `voted`. Now some of the logic of voting (not adding votes from the same participant twice) is removed from this module and pushed to somewhere else. I'm not sure this is best from a code perspective. I would rather still keep track of who voted in this struct. Maybe we could do something like make this a `HashMap<Addr,bool>` and initialize all of the participants to `false`? We could use that same struct to track who the participants are as well, instead of using the `Snapshot`

Reply:
good suggestion, I'll try this out

Reply:
changed to this:
```
#[cw_serde]
pub struct Participation {
    pub weight: nonempty::Uint256,
    pub voted: bool,
}

#[cw_serde]
pub struct WeightedPoll {
    poll_id: PollID,
    quorum: nonempty::Uint256,
    expires_at: u64,
    poll_size: u64,
    votes: Vec<Uint256>, // running tally of weighted votes
    status: PollStatus,
    participation: HashMap<String, Participation>,
}
```

now everything remains in the voting.rs